pkg/scheduler/objects/application.go

```diff
@@ -1782,3 +1782,10 @@ func (sa *Application) SetTimedOutPlaceholder(taskGroupName string, timedOut int
 		sa.placeholderData[taskGroupName].TimedOut = timedOut
 	}
 }
+
+// test only
+func (sa *Application) addPlaceholderDataWithLocking(ask *AllocationAsk) {
+	sa.Lock()
+	defer sa.Unlock()
+	sa.addPlaceholderData(ask)
+}
```

Comment:
Move this function into application_test.go since it is test-only. It's in the same package so will still compile.

Reply:
Done